dev/AnimatedVisualPlayer/TestUI/AnimatedVisualPlayerPage.xaml.cs

```diff
@@ -12,45 +13,70 @@
 using Windows.UI;
 using Windows.UI.Composition;
 using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Controls;
 using Windows.UI.Xaml.Hosting;
 using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Tests.MUXControls.ApiTests;
 
 namespace MUXControlsTestApp
 {
-    public class FallbackGrid : Windows.UI.Xaml.Controls.Grid
+    public sealed class FallbackGrid : Grid
     {
+        internal const int RectangleWidth = 100;
+        internal const int RectangleHeight = 100;
+        internal static readonly Color RectangleColor = Colors.Red;
+        // Gets the most recently created FallbackGrid
+        internal static FallbackGrid Latest;
+
         public FallbackGrid()
         {
-            _fallbackGrid = this;
-
+            Latest = this;
+            //Background = new SolidColorBrush(Colors.Red);
+            //this.HorizontalAlignment = HorizontalAlignment.Stretch;
+            //this.VerticalAlignment = VerticalAlignment.Stretch;
             Windows.UI.Xaml.Shapes.Rectangle rect = new Windows.UI.Xaml.Shapes.Rectangle();
-            rect.MinWidth = 100;
-            rect.MinHeight = 100;
-            rect.MaxWidth = 100;
-            rect.MaxHeight = 100;
-            rect.Fill = new SolidColorBrush(Colors.Red);
+            rect.MaxWidth = rect.MinWidth = RectangleWidth;
+            rect.MaxHeight = rect.MinHeight = RectangleHeight;
+            rect.Fill = new SolidColorBrush(RectangleColor);
             Children.Add(rect);
         }
-
-        static public FallbackGrid _fallbackGrid = null;
     }
 
-    /// <summary>
-    /// An empty page that can be used on its own or navigated to within a Frame.
-    /// </summary>
     [TopLevelTestPage(Name = "AnimatedVisualPlayer", Icon = "Animations.png")]
     public sealed partial class AnimatedVisualPlayerPage : TestPage
     {
-        private Visual _visual;
-
         public AnimatedVisualPlayerPage()
         {
             this.InitializeComponent();
+            SetAutomationPeerNames(this);
+        }
+
+        // Set the AutomationPeer name to be the same as the x:Name on each FrameworkElement.
+        // This is so we don't need to set both properties on the elements in XAML.
+        void SetAutomationPeerNames(FrameworkElement root)
+        {
+            var count = VisualTreeHelper.GetChildrenCount(root);
+            for (var i = 0; i < count; i++)
+            {
+                var child = VisualTreeHelper.GetChild(root, i);
+                if (child is FrameworkElement element)
+                {
+                    // Copy the x:Name into the AutomationPeer name.
+                    var name = element.Name;
+                    if (!string.IsNullOrEmpty(name))
```

Comment:
Optional: consider adding a check for an existing AutomationPeer Name before over-writing.

Reply:
Done.